File: 198-HouseRobber/198-HouseRobber.py
# Last updated: 08/05/2026, 09:31:08
class Solution:
    def rob(self, nums: List[int]) -> int:
        
        rob1=0

        rob2=0
        #[rob1,rob2,n, n+1, n+2]

        for x in nums:
            temp=max(rob1+x, rob2)
            rob1=rob2
            rob2=temp
            
        return temp

File: 198-HouseRobber/198-HouseRobber.cpp
// Last updated: 08/05/2026, 09:31:03
class Solution {
public:
    int rob(vector<int>& nums) {

        int rob1=0;
        int rob2=0;
        int temp=0;

        for(auto x : nums){
            temp= max(rob1+x, temp);
            rob1=rob2;
            rob2= temp;
        }
        
        
        return temp;
        
    }
};